core/server/worker/src/main/java/alluxio/worker/block/TieredBlockStore.java

```diff
@@ -219,7 +230,11 @@ public BlockReader getBlockReader(long sessionId, long blockId, long lockId)
     mLockManager.validateLock(sessionId, blockId, lockId);
     try (LockResource r = new LockResource(mMetadataReadLock)) {
       BlockMeta blockMeta = mMetaManager.getBlockMeta(blockId);
-      return new StoreBlockReader(sessionId, blockMeta);
+      BlockReader reader = new StoreBlockReader(sessionId, blockMeta);
+      if (TIMEOUT_DURATION > 0) {
```

Comment:
The PR is indicating that this timeout is only used for caching request. But it's actually used on all block reader. So if a positive TIMEOUT_DURATION is given, we will effectively create a thread pool for all readers created. Have we tested this feature in any ways? This PR doesn't include any test case. Are we sure there won't be any regression here?